Description: Fixed crash on exit
 This patch was backported from version 2.2.7 which is not yet released.
Bug-Ubuntu: https://launchpad.net/bugs/944675
Author: Andrey Karlov <dein.negativ@gmail.com>
Last-Update: 2012-04-10

--- a/eiskaltdcpp-qt/src/ArenaWidgetManager.cpp
+++ b/eiskaltdcpp-qt/src/ArenaWidgetManager.cpp
@@ -23,6 +23,8 @@ ArenaWidgetManager::ArenaWidgetManager() : QObject(NULL) {
 
 ArenaWidgetManager::~ArenaWidgetManager(){
     DEBUG_BLOCK
+
+    disconnect(this, 0, 0, 0);
     
     foreach ( ArenaWidget *awgt , widgets ) {
         if (dcpp::ISingleton *isingleton = dynamic_cast<dcpp::ISingleton*>(awgt)){
--- a/eiskaltdcpp-qt/src/GlobalTimer.cpp
+++ b/eiskaltdcpp-qt/src/GlobalTimer.cpp
@@ -21,7 +21,9 @@ GlobalTimer::GlobalTimer() : QObject(NULL), timer(new QTimer()), tickCount(0) {
 }
 
 GlobalTimer::~GlobalTimer() {
+    timer->stop();
 
+    disconnect(this, 0, 0, 0);
 }
 
 void GlobalTimer::slotTick() {
--- a/eiskaltdcpp-qt/src/main.cpp
+++ b/eiskaltdcpp-qt/src/main.cpp
@@ -203,6 +203,8 @@ int main(int argc, char *argv[])
 #ifdef USE_JS
     ScriptEngine::deleteInstance();
 #endif
+
+    GlobalTimer::deleteInstance();
     
     ArenaWidgetManager::deleteInstance();
     
@@ -211,9 +213,8 @@ int main(int argc, char *argv[])
     MainWindow::deleteInstance();
 
     WulforUtil::deleteInstance();
+
     WulforSettings::deleteInstance();
-    
-    GlobalTimer::deleteInstance();
 
     dcpp::shutdown();
 
